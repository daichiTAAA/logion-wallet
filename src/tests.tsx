import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
