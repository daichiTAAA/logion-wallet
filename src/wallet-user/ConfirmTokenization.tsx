import React from 'react';

export default function ConfirmTokenization() {
    return (
        <p>
            Your request has been sent to your legal officer.
        </p>
    )
}
