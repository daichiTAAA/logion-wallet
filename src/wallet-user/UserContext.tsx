import React, {useContext, useEffect, useState} from "react";
import {CreateTokenRequest, createTokenRequest as modelCreateTokenRequest} from "./Model";
import {TokenizationRequest} from "../legal-officer/Model";

export interface UserContext {
    legalOfficerAddress: string,
    userAddress: string,
    createTokenRequest: ((request: CreateTokenRequest) => Promise<TokenizationRequest>) | null,
    createdTokenRequest: TokenizationRequest | null
}

function initialContextValue(legalOfficerAddress: string, userAddress: string): UserContext {
    return {
        legalOfficerAddress,
        userAddress,
        createTokenRequest: null,
        createdTokenRequest: null
    }
}

const UserContextObject: React.Context<UserContext> = React.createContext(initialContextValue("", ""));

export interface Props {
    legalOfficerAddress: string,
    userAddress: string,
    children: JSX.Element | JSX.Element[] | null
}

export function UserContextProvider(props: Props) {
    const [contextValue, setContextValue] = useState<UserContext>(initialContextValue(props.legalOfficerAddress, props.userAddress));

    useEffect(() => {
        if (contextValue.createTokenRequest === null) {
            const createTokenRequest = async (request: CreateTokenRequest): Promise<TokenizationRequest> => {
                const createdTokenRequest = await modelCreateTokenRequest(request);
                setContextValue({...contextValue, createdTokenRequest})
                return createdTokenRequest;
            }
            setContextValue({...contextValue, createTokenRequest});
        }
    }, [contextValue, setContextValue])
    return (
        <UserContextObject.Provider value={contextValue}>
            {props.children}
        </UserContextObject.Provider>
    );
}

export function useUserContext(): UserContext {
    return {...useContext(UserContextObject)};
}
