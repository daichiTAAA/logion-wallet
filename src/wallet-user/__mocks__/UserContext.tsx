import {DEFAULT_LEGAL_OFFICER, TokenizationRequest} from "../../legal-officer/Model";
import {TEST_WALLET_USER} from "../Model.test";

export let createTokenRequest = () => null;

export let createdTokenRequest: TokenizationRequest | null = null;

export function useUserContext() {
    return {
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER,
        userAddress: TEST_WALLET_USER,
        createTokenRequest,
        createdTokenRequest
    };
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setCreatedTokenRequest(request: TokenizationRequest) {
    createdTokenRequest = request;
}
