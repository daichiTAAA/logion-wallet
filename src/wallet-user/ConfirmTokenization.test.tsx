jest.mock('./UserContext')

import React from "react";
import {setCreatedTokenRequest} from "./UserContext";
import {render, shallowRender} from "../tests";
import ConfirmTokenization from "./ConfirmTokenization";


test("renders empty", () => {
    const tree = render(<ConfirmTokenization/>)
    expect(tree).toMatchSnapshot();
});

test("renders confirmation", () => {
    setCreatedTokenRequest({
        id: "123456",
        legalOfficerAddress: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        requestedTokenName: "TOKEN1",
        bars: 1
    })
    const tree = render(<ConfirmTokenization/>)
    expect(tree).toMatchSnapshot();
})
