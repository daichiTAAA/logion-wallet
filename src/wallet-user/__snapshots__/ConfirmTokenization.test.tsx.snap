// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders confirmation 1`] = `
<p>
  Your request (
  123456
  ) has been sent to your legal officer.
</p>
`;

exports[`renders empty 1`] = `null`;
